mmdeploy/codebase/mmrotate/deploy/rotated_detection_model.py

```diff
@@ -0,0 +1,198 @@
+        device (str): A string represents device type.
+        deploy_cfg (str | mmcv.Config): Deployment config file or loaded Config
+            object.
+        model_cfg (str | mmcv.Config): Model config file or loaded Config
+            object.
+    """
+
+    def __init__(
+        self,
+        backend: Backend,
+        backend_files: Sequence[str],
+        class_names: Sequence[str],
+        device: str,
+        deploy_cfg: Union[str, mmcv.Config] = None,
+        model_cfg: Union[str, mmcv.Config] = None,
+    ):
+        super(End2EndModel, self).__init__(deploy_cfg=deploy_cfg)
+        model_cfg, deploy_cfg = load_config(model_cfg, deploy_cfg)
+        self.CLASSES = class_names
+        self.deploy_cfg = deploy_cfg
+        self.device = device
+        self.show_score = False
+        self._init_wrapper(
+            backend=backend, backend_files=backend_files, device=device)
+
+    def _init_wrapper(self, backend: Backend, backend_files: Sequence[str],
+                      device: str):
+        """Initialize the wrapper of backends.
+
+        Args:
+            backend (Backend): The backend enum, specifying backend type.
+            backend_files (Sequence[str]): Paths to all required backend files
+                (e.g. .onnx' for ONNX Runtime, '.param' and '.bin' for ncnn).
+            device (str): A string represents device type.
+        """
+        output_names = self.output_names
+        self.wrapper = BaseBackendModel._build_wrapper(
+            backend=backend,
+            backend_files=backend_files,
+            device=device,
+            input_names=[self.input_name],
+            output_names=output_names,
+            deploy_cfg=self.deploy_cfg)
+
+    def forward(self, img: Sequence[torch.Tensor],
+                img_metas: Sequence[Sequence[dict]], *args, **kwargs) -> list:
+        """Run forward inference.
+
+        Args:
+            img (Sequence[torch.Tensor]): A list contains input image(s)
+                in [N x C x H x W] format.
+            img_metas (Sequence[Sequence[dict]]): A list of meta info for
+                image(s).
+
+        Returns:
+            list: A list contains predictions.
+        """
+        input_img = img[0].contiguous()
+        img_metas = img_metas[0]
+        outputs = self.forward_test(input_img, img_metas, *args, **kwargs)
+        batch_dets, batch_labels = outputs[:2]
+        batch_size = input_img.shape[0]
+        rescale = kwargs.get('rescale', False)
+
+        results = []
+
+        for i in range(batch_size):
+            dets, labels = batch_dets[i], batch_labels[i]
+            if rescale:
+                scale_factor = img_metas[i]['scale_factor']
+
+                if isinstance(scale_factor, (list, tuple, np.ndarray)):
+                    assert len(scale_factor) == 4
+                    scale_factor = np.array(scale_factor)[None, :]  # [1,4]
+                scale_factor = torch.from_numpy(scale_factor).to(
```

Comment:
When using onnxruntime-gpu and run test.py with --device=cuda, the scale_factor is on cuda but dets is on cpu, so `dets[:, :4] /= scale_factor` will throw an error.